src/libxively/tls/xi_tls_layer.c

```diff
@@ -275,18 +280,6 @@ static xi_state_t recv_handler( void* context, void* data, xi_state_t in_out_sta
         return XI_STATE_OK;
     }
 
-    /* if we suppose to reuse the buffer */
-    if ( XI_STATE_WANT_READ == in_out_state && NULL != data_desc )
-    {
```

Comment:
Ok, so the story begins with the xi_mqtt_codec_layer - the xi_mqtt_codec_layer, before that PR change, could call the xi_tls_layer::pull function in order to notify it that it requires more data. In case if the parser couldn't finalise reading a message. After I added your patch plus checks for pending data to xi_tls_layer xi_mqtt_codec_layer does no longer need to do this, becuase the xi_tls_layer is self aware of the situation. In fact that xi_mqtt_codec_layer demand of more data was an overuse because of the lack of other solution that's why it has been taken out. So by taking that change out - we do no longer need a to check for this condition within the xi_tls_layer that's why we can remove that code.